public/app/core/logs_model.ts

```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
   let visibleRange = absoluteRange;
   let resolutionIntervalMs = intervalMs;
   let bucketSize = Math.max(resolutionIntervalMs * pxPerBar, minimumBucketSize);
+  let timeSpanOfLogs;
+  let timeSpanOfRange;
   // Clamp time range to visible logs otherwise big parts of the graph might look empty
   if (absoluteRange) {
-    const earliest = sortedRows[0].timeEpochMs;
-    const latest = absoluteRange.to;
-    const visibleRangeMs = latest - earliest;
+    const earliestTsLogs = sortedRows[0].timeEpochMs;
+    const latestTsLogs = sortedRows[sortedRows.length - 1].timeEpochMs;
+    const earliestTsAbsolute = absoluteRange.from;
+    const latestTsAbsolute = absoluteRange.to;
+
+    // Calculate timespan for received logs and range. This is used to calculate timespan coverage meta info of received logs.
+    // We are rounding this as some ds are rounding range in their requests which resulted in incorrect coverage (in edge cases).
+    timeSpanOfLogs = Math.ceil((latestTsLogs - earliestTsLogs) / 1000);
+    timeSpanOfRange = Math.ceil((latestTsAbsolute - earliestTsAbsolute) / 1000);
+
+    const visibleRangeMs = latestTsAbsolute - earliestTsLogs;
```

Comment:
Could it be renamed to `selectedRangeMs`? If I understand correctly it's not the range that is visible on the graph but the range that was selected for the query.

Reply:
This is actually visible range on the Graph, cause you are showing `latestTsAbsolute` and `earliestTsLogs`. It is called visible range across couple of places so I wouldn't change it. However, this is a good point - that instead of using actual log range, we  should use visible range. I'll change it in solution - which will simplify code a bit.